packages/cactus-plugin-ubiquity/src/main/typescript/ubiquity.ts

```diff
@@ -0,0 +1,227 @@
+    const authToken = process.env.UBIQUITY_AUTH_TOKEN;
+    if (!authToken) {
+      throw new Error("Auth token not defined");
+    }
+    this.authToken = authToken;
+    Checks.truthy(opts, `${fnTag} arg options`);
+    Checks.truthy(opts.authToken, `${fnTag} options.authToken`);
+    Checks.truthy(opts.logLevel, `${fnTag} options.logLevel`);
+    Checks.truthy(opts.instanceId, `${fnTag} options.instanceId`);
+    this.log = LoggerProvider.getOrCreate({
+      label: this.className,
+      level: opts.logLevel,
+    });
+
+    this.instanceId = opts.instanceId || uuidV4();
+    this.log.debug("Initialized");
+    this.client = new UbiquityClient(
+      opts.authToken,
+      opts.basePath || "https://ubiquity.api.blockdaemon.com/v1",
+    );
+    this.prometheusExporter =
+      opts.prometheusExporter ||
+      new PrometheusExporter({ pollingIntervalInMin: 1 });
+
+    this.prometheusExporter.startMetricsCollection();
+  }
+
+  // https://ubiquity.docs.blockdaemon.com/swagger-ui/#/Accounts/v2GetTxsByAddress
+  public async GetTxsByAddress(
+    protocol: PROTOCOL_TYPE,
+    network: NETWORK_TYPE,
+    address: string,
+    continuation?: string,
+    limit?: number,
+  ): Promise<TxPage> {
+    const fnTag = `${this.className}:GetTxsByAddress`;
+    this.log.debug("enter ", fnTag);
+    const response = await this.client.accountsApi.getTxsByAddress(
+      protocol,
+      network,
+      address,
+      "asc",
+      continuation,
+      limit,
+    );
+    this.prometheusExporter.addMethodCall();
+    return response.data;
+  }
+
+  // https://ubiquity.docs.blockdaemon.com/swagger-ui/#/Accounts/GetBalancesByAddresses
+  public GetBalancesByAddresses(
+    protocol: PROTOCOL_TYPE,
+    network: NETWORK_TYPE,
+    address: string,
+  ): Promise<AxiosResponse<Balance[]>> {
+    const fnTag = `${this.className}:GetBalancesByAddresses`;
+    this.log.debug("enter ", fnTag);
+    const accounts = this.client.accountsApi.getListOfBalancesByAddress(
+      protocol,
+      network,
+      address,
+    );
+    this.prometheusExporter.addMethodCall();
+    return accounts;
+  }
+
+  // https://ubiquity.docs.blockdaemon.com/swagger-ui/#/Transactions/V2GetTx
+  public GetTx(
+    protocol: PROTOCOL_TYPE,
+    network: NETWORK_TYPE,
+    id: string,
+  ): Promise<AxiosResponse<Tx>> {
+    const fnTag = `${this.className}:GetTx`;
+    this.log.debug("enter ", fnTag);
+
+    const transactions = this.client.transactionsApi.getTx(
+      protocol,
+      network,
+      id,
+    );
+    this.prometheusExporter.addMethodCall();
+    return transactions;
+  }
+
+  // https://ethereum.org/en/developers/docs/apis/json-rpc/
+  public async nativeGetTx(
+    protocol: PROTOCOL_TYPE,
+    network: NETWORK_TYPE,
+  ): Promise<nodeFetch.Response> {
+    const fnTag = `${this.className}:NativeGetTx`;
+    this.log.debug("enter ", fnTag);
+    const ethereumUrl = `https://svc.blockdaemon.com/${protocol}/${network}/native`;
```

Comment:
Will the URL's strictly start with `svc.blockdaemon.com` prefix?

Reply:
Yes!